db/records.py

```diff
@@ -299,23 +300,30 @@ def create_records_from_csv(
         escape=None,
         quote=None,
 ):
-    with open(csv_filename, 'rb') as csv_file:
+    with open(csv_filename, "rb") as csv_file:
         with engine.begin() as conn:
             cursor = conn.connection.cursor()
-            relation = '.'.join('"{}"'.format(part) for part in (table.schema, table.name))
-            formatted_columns = '({})'.format(','.join([f'"{column_name}"' for column_name in column_names]))
-
-            copy_sql = f'COPY {relation} {formatted_columns} FROM STDIN CSV'
-            if header:
-                copy_sql += " HEADER"
-            if delimiter:
-                copy_sql += f" DELIMITER E'{delimiter}'"
-            if escape:
-                copy_sql += f" ESCAPE '{escape}'"
-            if quote:
-                if quote == "'":
-                    quote = "''"
-                copy_sql += f" QUOTE '{quote}'"
+            relation = sql.SQL(".").join(
+                sql.Identifier(part) for part in (table.schema, table.name)
+            )
+            formatted_columns = sql.SQL(",").join(
+                sql.Identifier(column_name) for column_name in column_names
+            )
+
+            copy_sql = sql.SQL(
+                "COPY {relation} ({formatted_columns}) FROM STDIN CSV {header} {delimiter} {escape} {quote}"
+            ).format(
+                relation=relation,
+                formatted_columns=formatted_columns,
+                header=sql.SQL("HEADER" if header else ""),
```

Comment:
Do we need to call `sql.SQL` on all the format strings since we're already calling it on the main string?

Reply:
Yes, we need to recall `sql.SQL` for each component, because sql.SQL is not considered as a `String` but a `sql. Composable` which means we're not allowed to merge these two different types.

For example, when I tried to remove sql.SQL from the header, I received this warning, meanwhile, the program will crash.

![Screenshot](https://user-images.githubusercontent.com/15726680/130507329-071945e2-540c-41b9-871b-1955574085da.png)